spec/models/admin_user_spec.rb

```diff
@@ -0,0 +1,22 @@
+describe AdminUser do
+  it 'return type should be User' do
+    testuser = AdminUser.find_or_create
+    expect(testuser).to be_a_kind_of(User)
+  end
+  context 'no existing admin' do
+    it 'first_name and last_name should equal to the specified values' do
+      testuser = AdminUser.find_or_create
+      expect(testuser.first_name).to eq('admin')
+      expect(testuser.last_name).to eq('admin')
+    end
+  end
+  context 'existing admin' do
+    it 'first_name and last_name should ot be changed for existing admin' do
```

Comment:
In addition to my comments above, there's a typo with "ot". And once again, make sure the sentence makes grammatical sense.

Reply:
Also make sure to split these up into separate tests as well.

You can use a `let!(:admin_user)` in this context block to ensure an admin user is already created at each test (you can check out the documentation for the difference between `let` and `let!`).